### PR TITLE
Improve settings migration and management

### PR DESCRIPTION
**Pull request by kilrah**
Users now have a decent way to remove settings during uninstall, and import from older version is no more mandatory.

### CODE REVIEW

**Comment by mpaperno** [comment]
You should be able to translate by wrapping the string in `QCoreApplication::translate()` function.

Though personally I'd avoid using GUI/interactive functions in a non-GUI class like AppData altogether.

**Comment by kilrah** [comment]
The translate comment was about the NSIS uninstaller, not Qt

Agreed about the location, but I haven't been able to figure out how to do it cleanly...

**Comment by bsongis** [comment]
I would split `importSettings` into 2 functions: the first one for checking if we can import some settings, and the second to do the job, so that the confirmation message box thing is not in this file

**Comment by kilrah** [comment]
Thanks for the hint

**Comment by bsongis** [comment]
Perhaps you could remove the whole part of code which crashes, it's so old ...

File: companion/src/companion.cpp [hunk 1]
```diff
@@ -71,6 +71,20 @@ int main(int argc, char *argv[])
 
   CustomDebug::setFilterRules();
 
+  if (!g.hasCurrentSettings()) {
+    QString previousVersion;
+    if (g.findPreviousVersionSettings(&previousVersion)) {
+       QMessageBox msgBox;
+       msgBox.setText("We have found existing settings for Companion version: " + previousVersion + ".\nDo you want to import them?");
```
Comment (bsongis) [inline companion/src/companion.cpp]: No translations?

**Review by bsongis — APPROVED** [review]

File: companion/src/companion.cpp (hunk 1 above)
Comment (kilrah) [reply companion/src/companion.cpp]: Ah yes forgot that...

**Comment by kilrah** [comment]
> Perhaps you could remove the whole part of code which crashes, it's so old ...

Remove the whole import from c9x then? Cause it'll likely be pretty broken without that...

**Comment by bsongis** [comment]
👍

**Pull request merged**